src/cm/commands/getFile/getFile.ts

```diff
@@ -0,0 +1,64 @@
+import { dirname, join, relative } from "path";
+import { existsSync, promises } from "fs";
+import { ICmParser, ICmResult, ICmShell } from "../../shell";
+import { GetFileParser } from "./getFileParser";
+import { Uri } from "vscode";
+
+export class GetFile {
+  public static cachedFileLocation(
+      rootDir: string,
+      filePath: Uri,
+      changeset: number
+  ): Uri {
+    const cacheDir = join(rootDir, ".plastic", "fileCache");
+    const outputFile = join(cacheDir, changeset.toString(), relative(rootDir, filePath.fsPath));
+    return Uri.file(outputFile);
+  }
+
+  public static async run(
+      rootDir: string,
+      filePath: Uri,
+      changeset: number,
+      shell: ICmShell
+  ): Promise<Uri | undefined> {
+    const parser: ICmParser<void> = new GetFileParser();
+
+    if (filePath.fsPath.includes(".plastic")) {
+      return undefined;
+    }
+
+    const fileSpec = `${filePath.fsPath}#cs:${changeset}`;
+    const cacheDir = join(rootDir, ".plastic", "fileCache");
+    const outputFile = join(cacheDir, changeset.toString(), relative(rootDir, filePath.fsPath));
+
+    // prune old changesets
+    const cacheDirContents = await promises.readdir(cacheDir);
+    for (const file of cacheDirContents) {
+      const fileNameAsChangeset = parseInt(file, 10);
+      if (!isNaN(fileNameAsChangeset) && fileNameAsChangeset < changeset) {
+        await promises.rmdir(join(cacheDir, file), { recursive: true });
+      }
+    }
+
+    if (!existsSync(dirname(outputFile))) {
```

Comment:
This check needs to take place before the call to `promises.readdir(cacheDir)` to ensure that `cacheDir` exists. I'll probably merge this PR and submit another one to have it ready ASAP, to prevent #311 from getting stuck :)

Reply:
If @seesemichaelj could grant me permission to push to this PR (not sure if that's feasible) I have a commit ready to go 😄

Reply:
Thanks a lot!